connector-email/src/main/java/com/redhat/cloud/notifications/connector/email/processors/it/ITResponseProcessor.java

```diff
@@ -0,0 +1,58 @@
+package com.redhat.cloud.notifications.connector.email.processors.it;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.cloud.notifications.connector.email.constants.ExchangeProperty;
+import com.redhat.cloud.notifications.connector.email.processors.it.pojo.response.ITUserResponse;
+import org.apache.camel.Exchange;
+import org.apache.camel.Processor;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@ApplicationScoped
+public class ITResponseProcessor implements Processor {
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    /**
+     * Processes the response from the IT service. Grabs the usernames from the
+     * response.
+     * @param exchange the exchange of the pipeline.
+     * @throws JsonProcessingException if the incoming payload cannot be read
+     * into the {@link ITUserResponse} model.
+     */
+    @Override
+    public void process(final Exchange exchange) throws JsonProcessingException {
+        final String body = exchange.getMessage().getBody(String.class);
+        final List<ITUserResponse> itUserResponses = Arrays.asList(this.objectMapper.readValue(body, ITUserResponse[].class));
+
+        // Get ready the set of usernames that we will transform from the
+        // incoming payload. If multiple pages are fetched, this might get
+        // called multiple times, that is why we need to be safe.
+        Set<String> usernames = exchange.getProperty(ExchangeProperty.USERNAMES, Set.class);
+        if (usernames == null) {
+            usernames = new HashSet<>();
+            exchange.setProperty(ExchangeProperty.USERNAMES, usernames);
+        }
```

Comment:
Can this really be executed since `DispatcherProcessor` creates the `HashSet` instance?

Reply:
Nope! Removed!